sql/core/src/test/scala/org/apache/spark/sql/connector/MergeIntoTableSuiteBase.scala

```diff
@@ -2098,6 +2103,7 @@ abstract class MergeIntoTableSuiteBase extends RowLevelOperationSuiteBase
 
       val table = catalog.loadTable(ident)
       val commitProps = table.asInstanceOf[InMemoryTable].commits.last.properties
+      assert(commitProps("merge.numSourceRows") === "4")
```

Comment:
let's run these tests in AQE/non-AQE mode to verify the walk works in both scenarios.